### PR TITLE
Added Metadata to Slot & CG in DSSCollection

### PR DESCRIPTION
**Pull request by HouseOfHufflepuff**
Added metadata as requested by collection-group team. Metadata is used to sync the on-chain and off-chain creation of a collection group and slot.

### CODE REVIEW

**Review by pickerflicker — DISMISSED** [review]

File: dss-collection-nft/contracts/DSSCollection.cdc [hunk 1]
```diff
@@ -164,7 +166,8 @@ pub contract DSSCollection: NonFungibleToken {
             collectionGroupID: UInt64,
             logicalOperator: String,
             required: Bool,
-            typeName: Type
+            typeName: Type,
+            batchKey: String
```
Comment (judezhu) [inline dss-collection-nft/contracts/DSSCollection.cdc:170]: Thanks Jer. `batchKey` is internal technical terms we used in dapper. Should we use `UUID` or `LinkID` to make it more generic?

Comment (HouseOfHufflepuff) [reply dss-collection-nft/contracts/DSSCollection.cdc:170]: Sure. I have no preference on the name. Will change it in a sec.

Comment (HouseOfHufflepuff) [reply dss-collection-nft/contracts/DSSCollection.cdc]: Changed to metadata: {String: String}

**Review by Deewai — APPROVED** [review]

**Review by judezhu — APPROVED** [review]
Thanks, Jer!

**Pull request merged**